dataplicity/m2m/bencode.py

```diff
@@ -60,12 +61,19 @@ def encode(obj):
 
     def add_encode(obj):
         if isinstance(obj, bytes):
-            append(b"{}:{}".format(len(obj), obj))
+            if PY2:
+                append(b"{}:{}".format(len(obj), obj))
+            elif PY3:
+                append(six.b("{}:".format(len(obj))))
+                append(obj)
         elif isinstance(obj, text_type):
             obj = obj.encode('utf-8')
-            append(b"{}:{}".format(len(obj), obj))
+            if PY2:
+                append(b"{}:{}".format(len(obj), obj))
+            elif PY3:
+                add_encode(obj)
```

Comment:
Only once in the agent. There is similar code in m2mclient. But that's not encumbered with the need to be both PY2 and PY3 compatible.